protos/controlplane.proto

```diff
@@ -44,10 +44,16 @@ message Rule {
 }
 
 message Stream {
+  message Keyed {
+    google.protobuf.Duration ttl = 1;
+    int32 max_keys = 2;
+  }
+
   string uid = 1;
   string name = 2;
   Rule rule = 3;
   bool export_raw_samples = 4;
+  Keyed keyed = 5;
```

Comment:
Implicitly define a stream as keyed if these configuration options are set feels like it could catch us by surprise. Actually, I imagine not even the options are required, as long as the `keyed` object is not nil it is considered that the stream has been configured as keyed.
I think that adding a boolean inside the `Keyed` message or have some other field at the `Stream` level that defines the stream as keyed (like an enum that defines the type of stream) and then have the options in `keyed` would be more explicit.

Reply:
Mh you are right that we are following a similar pattern there. What makes me think that in this case makes it feel more unexpected is that setting the option `ttl` and `max_keys` doesn't really tell us if the stream is configured as keyed or not it just configures some options that may be seen as not required of a keyed stream.

In the other cases setting `limit` or the `oneof Sampling` field makes it clear to me that there is a limit or a sampling strategy applied.